streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -425,12 +449,14 @@ public void flush() {
 
     /**
      * {@link StateStore#close() Close} all stores (even in case of failure).
-     * Log all exception and re-throw the first exception that did occur at the end.
+     * Log all exceptions and re-throw the first exception that occurred at the end.
      *
      * @throws ProcessorStateException if any error happens when closing the state stores
      */
     @Override
     public void close() throws ProcessorStateException {
+        unregisterAllStoresWithChangelogReader();
+
```

Comment:
nit: I think now it's better to move the debug line 464 before `unregisterAllStoresWithChangelogReader`.